### PR TITLE
Bugfix/sdp rc

### PR DESCRIPTION
**Pull request by kamil-stasiak**
To avoid Race Conditions, calls to functions such as addTrack, removeTrack, and replaceTrack will be queued. At most, one task of this type can be executed at the same time. To initiate the next task, the appropriate conditions must be met.

The processing flag indicates that the processing of sdpOffer and sdpAnswer is ongoing.

The conditions required to start processing the next task from the queue:
```ts
if (this.processing) return;
    if (
      this.connection &&
      (this.connection.signalingState !== "stable" ||
        this.connection.connectionState !== "connected" ||
        this.connection.iceConnectionState !== "connected")
    )
      return;
// Start task processing
```

The test `RC: Quickly add and remove a track` has been skipped because I suspect that a Race Condition occurs in Jellyfish itself or the rtc-engine.

### CODE REVIEW

File: e2e/scenarios/raceCondition.spec.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      await takeScreenshot(page, testInfo, "After test Screenshot");
+    });
+  }
+});
+
+/*
+ * Test in this file should be run a few times in a row to be sure that there is no race conditions.
+ * To run a test multiple times use command:
+ *
+ * npm run e2e -- --repeat-each=20 -g "Test name"
```
Comment (szebniok) [inline e2e/scenarios/raceCondition.spec.ts:30]: https://playwright.dev/docs/test-parameterize#parameterized-tests Couldn't we just express it with code?

**Review by szebniok — CHANGES_REQUESTED** [review]

**Review by mickel8 — APPROVED** [review]

File: e2e/scenarios/raceCondition.spec.ts (hunk 1 above)
Comment (kamil-stasiak) [reply e2e/scenarios/raceCondition.spec.ts:30]: For convenience, I've added `e2e:repeat` to package json with this command: `playwright test --repeat-each=20`

File: e2e/app/src/VideoPlayerWithDetector.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if (!videoElementRef.current) return;
+    videoElementRef.current.srcObject = stream || null;
+  }, [stream]);
+
+  useEffect(() => {
+    const id = setInterval(() => {
+      const videoElement = videoElementRef.current;
+      if (!videoElement || videoElement.videoWidth === 0) return;
+
+      const canvas = document.createElement("canvas");
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply e2e/app/src/VideoPlayerWithDetector.tsx:34]: I don't know how to capture the stream from `OffscreenCanvas`, and I don't want to spend more time on this task. However, OffscreenCanvas could be useful in the Dashboard. Thank you for this suggestion

File: e2e/scenarios/utils.ts [hunk 3]
```diff
@@ -91,3 +122,85 @@ export const createPeer = async (page: Page, roomId: string, enableSimulcast: bo
...
+
+export const addBothMockTracks = async (page: Page) =>
+  await test.step("Add both tracks", async () =>
+    await page
+      .getByRole("button", {
+        name: "Add both",
+        exact: true,
+      })
+      .click());
+
```
[the remark this answers is not in the document]
Comment (kamil-stasiak) [reply e2e/scenarios/utils.ts:162]: Done ✅

**Review by szebniok — APPROVED** [review]

**Pull request merged**